### PR TITLE
addition of test classes. Closes #586

### LINKED ISSUE #586: Add job request/response tests
For great examples see:

- [`DatasetRequestTest`](https://github.com/MarquezProject/marquez/blob/master/src/test/java/marquez/api/models/DatasetRequestTest.java)
- [`DatasetResponseTest`](https://github.com/MarquezProject/marquez/blob/master/src/test/java/marquez/api/models/DatasetResponseTest.java)
- [`DatasetsResponseTest`](https://github.com/MarquezProject/marquez/blob/master/src/test/java/marquez/api/models/DatasetsResponseTest.java)

### PR DESCRIPTION
**Pull request by ravikamaraj**
@wslulciuc , can you review. thanks!

Also please find below opportunity for  code cleanup, can you review and confirm
- CommonModelGenerator::newDatasetUrnWith, is not referenced/used
- CommonModelGenerator::newIdWithBound, bound argument is not used

### CODE REVIEW

File: src/test/java/marquez/api/models/JobRequestTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import marquez.common.models.DatasetUrn;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+@Category(UnitTests.class)
+public class JobRequestTest {
+  private static final List<String> INPUT_DATASET_URNS =
+      CommonModelGenerator.newDatasetUrns(4).stream().map(DatasetUrn::getValue).collect(toList());
+  private static final List<String> OUTPUT_DATASET_URNS =
+      CommonModelGenerator.newDatasetUrns(2).stream().map(DatasetUrn::getValue).collect(toList());
```
Comment (wslulciuc) [inline src/test/java/marquez/api/models/JobRequestTest.java]: _Minor_: Let's statically import `CommonModelGenerator.newDatasetUrns()` to simplify initialization to:

```
... = newDatasetUrns(4).stream().map(DatasetUrn::getValue).collect(toImmutableList());
```

**Comment by wslulciuc** [comment]
@ravikamaraj, great work! Just a minor comment on the use of `Collectors.toList`. We are moving towards the usage of [Guava](https://github.com/google/guava) for its immutable collection API.

In your PR, mind using `ImmutableList.toImmutableList`? For an example, see [`DatasetResponseMapper.map(List)`](https://github.com/MarquezProject/marquez/blob/master/src/main/java/marquez/api/mappers/DatasetResponseMapper.java#L39)

**Comment by wslulciuc** [comment]
> CommonModelGenerator::newDatasetUrnWith, is not referenced/used

Yep! You can remove 👍 

> CommonModelGenerator::newIdWithBound, bound argument is not used

This method is used [here](https://github.com/MarquezProject/marquez/blob/master/src/test/java/marquez/common/models/CommonModelGenerator.java#L63)

File: src/test/java/marquez/api/models/JobRequestTest.java (hunk 1 above)
Comment (ravikamaraj) [reply src/test/java/marquez/api/models/JobRequestTest.java]: done

**Comment by ravikamaraj** [comment]
> @ravikamaraj, great work! Just a minor comment on the use of `Collectors.toList`. We are moving towards the usage of [Guava](https://github.com/google/guava) for its immutable collection API.
> 
> In your PR, mind using `ImmutableList.toImmutableList`? For an example, see [`DatasetResponseMapper.map(List)`](https://github.com/MarquezProject/marquez/blob/master/src/main/java/marquez/api/mappers/DatasetResponseMapper.java#L39)

done

**Comment by ravikamaraj** [comment]
> > CommonModelGenerator::newDatasetUrnWith, is not referenced/used
> 
> Yep! You can remove 👍
> 
> > CommonModelGenerator::newIdWithBound, bound argument is not used
> 
> This method is used [here](https://github.com/MarquezProject/marquez/blob/master/src/test/java/marquez/common/models/CommonModelGenerator.java#L63)
`CommonModelGenerator::newDatasetUrnWith` - removed

`CommonModelGenerator::newIdWithBound` - Yes, the method is used, but  method argument `bound` is not used inside the method

**Comment by wslulciuc** [comment]
> CommonModelGenerator::newIdWithBound - Yes, the method is used, but method argument bound is not used inside the method

Ahh, right! Working on a PR to clean this up a bit.

**Pull request merged**